jsk_pcl_ros/src/line_segment_detector_nodelet.cpp

```diff
@@ -241,21 +245,55 @@ namespace jsk_pcl_ros
     pcl::PointIndices::Ptr rest_indices (new pcl::PointIndices);
     rest_indices->indices = indices->indices;
     // use RANSAC to segment lines
-    pcl::SACSegmentation<PointT> seg;
-    seg.setOptimizeCoefficients (true);
-    seg.setModelType (pcl::SACMODEL_LINE);
-    seg.setMethodType (pcl::SAC_RANSAC);
-    seg.setDistanceThreshold (outlier_threshold_);
-    seg.setMaxIterations (max_iterations_);
-    seg.setInputCloud(cloud);
+    if(segmentation_update_) {
+      seg_.setOptimizeCoefficients (true);
+      seg_.setModelType (pcl::SACMODEL_LINE);
+      switch (segmentation_method_) {
+        case jsk_pcl_ros::LineSegmentDetector_SAC_RANSAC: {
+          seg_.setMethodType(pcl::SAC_RANSAC);
```

Comment:
It seems that you can just set `seg_.setMethodType(segmentation_method_);`, seeing the documentation http://docs.pointclouds.org/trunk/method__types_8h_source.html. Is that possible?